src/data_source/storage/sql.rs

```diff
@@ -1787,6 +1789,636 @@ impl SqlStorage {
+                        JOIN payload AS p ON h.height = p.height
+                        WHERE h.hash = $1
+                        ORDER BY h.height DESC 
+                        LIMIT 1"
+                ),
+                vec![Box::new(hash.to_string())],
+            ),
+        };
+
+        let query_result = self.query_one(&query, params).await?;
+        let block = parse_block(query_result)?;
+
+        Ok(
+            explorer::data_source::BlockDetail::try_from(block).map_err(|e| QueryError::Error {
+                message: e.to_string(),
+            })?,
+        )
+    }
+
+    async fn get_transaction_summaries(
+        &self,
+        request: &explorer::data_source::GetTransactionSummariesRequest<Types>,
+    ) -> QueryResult<Vec<explorer::data_source::TransactionSummary<Types>>> {
+        let range = &request.range;
+        let target = &range.target;
+        let filter = &request.filter;
+
+        let (query, params, offset): (String, Vec<Box<dyn ToSql + Send + Sync>>, Option<usize>) =
+            match (target, filter) {
+                (_, explorer::data_source::TransactionSummaryFilter::RollUp(_)) => {
+                    // TODO: implement roll-up based filtering
+                    return Ok(vec![]);
+                }
+                (
+                    explorer::data_source::TransactionIdentifier::Latest,
+                    explorer::data_source::TransactionSummaryFilter::None,
+                ) => (
+                    format!(
+                        "SELECT {BLOCK_COLUMNS}
+                           FROM header AS h
+                           JOIN payload AS p ON h.height = p.height
+                           WHERE h.height IN (
+                                SELECT t1.block_height
+                                    FROM transaction AS t1
+                                    ORDER BY (t1.block_height, t1.index) DESC
+                                    LIMIT $1
+                            )
+                           ORDER BY h.height DESC"
+                    ),
+                    vec![Box::new(range.num_transactions.get() as i64)],
+                    Some(0),
+                ),
+                (
+                    explorer::data_source::TransactionIdentifier::Latest,
+                    explorer::data_source::TransactionSummaryFilter::Block(height),
+                ) => (
+                    format!(
+                        "SELECT {BLOCK_COLUMNS}
+                           FROM header AS h
+                           JOIN payload AS p ON h.height = p.height
+                           WHERE h.height IN (
+                                SELECT t1.block_height
+                                    FROM transaction AS t1
+                                    WHERE t1.block_height = $2
+                                    ORDER BY (t1.block_height, t1.index) DESC
+                                    LIMIT $1
+                            )
```

Comment:
This is true, and will change this.

The limit is meant to ensure that we only select the `block`s that we need, but since the `block` has a further constraint of just that single `block` itself the only returned value can possibly be just that single `block`.

Reply:
Since the queries on this have been reworked, this may no longer be applicable.